src/libsyntax/tokenstream.rs

```diff
@@ -437,30 +405,48 @@ pub fn cons_tt_ts(tt: TokenTree, ts: TokenStream) -> TokenStream {
 ///
 ///    `maybe_path_prefix("a::b::c(a,b,c).foo()") -> (a::b::c, "(a,b,c).foo()")`
 impl TokenStream {
-    pub fn respan(self, span: Span) -> TokenStream {
+
+    /// Convert a vector of `TokenTree`s into a `TokenStream`.
+    pub fn to_ts(trees: Vec<TokenTree>) -> TokenStream {
+        let span = covering_span(&trees);
         TokenStream {
-            tts: self.tts,
+            tts: trees,
             span: span,
         }
     }
 
-    pub fn as_slice(&self) -> &TokenSlice {
+    /// Given two `TokenStream`s, return one that contains both.
+    pub fn append_streams(mut self, ts2: TokenStream) -> TokenStream {
```

Comment:
and could ts2 be a TokenSlice rather than stream? And this should take &mut self and not return anything